trafilatura/xml.py

```diff
@@ -291,21 +295,33 @@ def process_element(element, returnlist, include_formatting):
             returnlist.extend(['![', text.strip(), ']', '(', element.get('src', ''), ')'])
         # newlines for textless elements
         if element.tag in NEWLINE_ELEMS:
-            returnlist.append('\n')
             # add line after table head
             if element.tag == "row":
-                num_cells = len(element.xpath("./cell[@role='head']"))
-                if num_cells > 0:
-                    returnlist.append("---|" * len(element.xpath(".//cell")) + "\n")
-        return  # Nothing more to do with textless elements
+                max_span = int(element.get("span", 1))
+                cell_count = len(element.xpath(".//cell"))
+                # row ended so draw extra empty cells to match max_span
+                returnlist.append("|" * (max_span - cell_count) + "\n")
+                # if this is a head row, draw the separator below
+                is_head = bool(element.xpath("./cell[@role='head']"))
+                if is_head:
+                    returnlist.append("\n" + "---|" * max_span + "\n")
+            else:
+                returnlist.append('\n')
+        if element.tag != 'cell':
+            # cells still need to append vertical bars
+            # but nothing more to do with other textless elements
+            return
 
     # Process text
 
     # Common elements (Now processes end-tag logic correctly)
+    within_cell = element.xpath("ancestor::cell")
     if element.tag == 'p' and include_formatting:
-        returnlist.append('\n\u2424\n')
+        if not within_cell:
```

Comment:
Maybe rather these conditions or would something go wrong?
`if element.tag == 'p' and include_formatting and not within_cell:`
(same below)

Reply:
The idea here was to terminate processing within the condition branch. For example if `element.tag == 'p' and include_formatting` is satsified then I enter the code below and no other `elif/else` branch will ever run.

If I inline it all as `if element.tag == 'p' and include_formatting and not within_cell:` then other branches may still run (which may not be desired).

So in case of `element.tag == 'p' and include_formatting`, we only ever run the couple of lines below and never fall back to any other condition like the `else` statement or other `elif`.

Reply:
OK, that makes sense.